packages/data-api/scripts/createAnalyticsTableIndexes.sql

```diff
@@ -1,16 +1,34 @@
 do $$ 
 declare
   tStartTime TIMESTAMP;
+  tAnalyticsIndexName TEXT := 'analytics_data_element_entity_date_idx';
+  tEventsIndexName TEXT := 'analytics_data_group_entity_event_date_idx';
 
 begin
   RAISE NOTICE 'Creating analytics table indexes...';
   
-  tStartTime := clock_timestamp();
-  CREATE INDEX IF NOT EXISTS analytics_data_element_entity_date_idx ON public.analytics(data_element_code, entity_code, date desc);
-  RAISE NOTICE 'Created [data_element, entity, date] index, took %', clock_timestamp() - tStartTime;
-  
-  tStartTime := clock_timestamp();
-  CREATE INDEX IF NOT EXISTS analytics_data_group_entity_event_date_idx ON public.analytics(data_group_code, entity_code, event_id, date desc);
-  RAISE NOTICE 'Created [data_group, entity, event, date] index, took %', clock_timestamp() - tStartTime;
+  IF (SELECT count(*) = 0
+    FROM pg_class c
+    WHERE c.relname = tAnalyticsIndexName 
+    AND c.relkind = 'i')
+  THEN
+    tStartTime := clock_timestamp();
+    PERFORM mv$addIndexToMv$Table(mv$buildAllConstants(), 'public', 'analytics', 'analytics_data_element_entity_date_idx', 'data_element_code, entity_code, date desc');
```

Comment:
I chose not to use `CREATE INDEX IF NOT EXISTS...` in the `mvrefresh` functions I added, perhaps I should have? Refer to that PR here: https://github.com/beyondessential/pg-mv-fast-refresh/pull/2

Reply:
That's what your query for whether the `pg_class` entry exists fits in though right? So I'm totally fine with it. Interested in the reason though?

Reply:
Yup, the whole code block basically just checks if an index with that name exists:
```
SELECT count(*) = 0
    FROM pg_class c
    WHERE c.relname = tAnalyticsIndexName 
    AND c.relkind = 'i'
```
So this script will handle the if exists/not exists case gracefully. Just was wondering about the `mv$addIndexToMv$Table` function in general, but I felt it might be a bit more consistent to not use `IF EXISTS` in that function